### PR TITLE
feat(stream): add project icons

### PR DESCRIPTION
**Pull request by Chrissy**
project-avi

Adds a project avatar to:
  * Issue list
  * Issue detail

### CODE REVIEW

File: src/sentry/static/sentry/app/components/platformList.jsx
Comment (Chrissy) [inline src/sentry/static/sentry/app/components/platformList.jsx]: this was already passed in from the child, but wasn't getting there because we weren't moving `className` through

**Comment by Chrissy** [comment]
resolves [SEN-42](https://getsentry.atlassian.net/jira/software/projects/SEN/boards/55?selectedIssue=SEN-42)

File: src/sentry/static/sentry/app/views/groupDetails/shared/header.jsx
Comment (markstory) [inline src/sentry/static/sentry/app/views/groupDetails/shared/header.jsx]: You could use `location` local variable here instead of reaching into window.

**Review by markstory — APPROVED** [review]
Looks good to me.

**Comment by benvinegar** [comment]
I still feel this could be tighter by just pairing the icon with the callsign (e.g. `JAVASCRIPT-123`)

**Comment by Chrissy** [comment]
@benvinegar I think you could be right, but I do like how explicit the extra column is. Let's try this for a bit and see how it feels. Do you imagine this combination being used in both views, or only issue details?

**Comment by benvinegar** [comment]
Both views / all views. It would be nice to just say "the callsign should be paired with the platform icon" and we have a generic way of presenting that throughout the app. What we're doing now requires a special treatment everywhere.

**Comment by benvinegar** [comment]
For example, the events view could show the callsign too (instead of _just_ project), which would offer additional value because you can see how individual events are grouped.

**Comment by Chrissy** [comment]
I'm on board. 

since this is more or less ready to go, I'm just gonna put it out, but I can follow up this week with a version that implements that idea.

**Comment by Chrissy** [comment]
@EvanPurkhiser passive aggressive thumbs do me no good please speak up

**Comment by EvanPurkhiser** [comment]
Oops, I thought I left my comment here, sorry!

It was a bit of a drive by review, but I was going to say I definitely agree with Ben, that it feels pretty redundant, and that the common case is single projects, so having a column dedicated for this is pretty redundant, and I would say shipping out a change just because it's done and then coming back to it later doesn't seem great.

**Comment by Chrissy** [comment]
I wasn't so much shipping it because it is done, as I am trying not to let this pull get too big and most of this code would be used in that implementation also. 

But I'll just roll it all together here and get a second review.

**Comment by Chrissy** [comment]
also while this idea _seems_ good to everyone, I'm not sure we are gonna love the results as much as we think we will. The issue list view is a compact view—I'm already having to really slim things down just to add the avatar—and putting something akin to a badge in there will cause layout challenges, especially at viewports sub 1300 pixels, and this doesn't factor in longer project names.

One easy addition would be to combine them on the project details view but skip it on the list for now. We have more space in the detail view and since the id is already there, it makes more sense.

**Comment by Chrissy** [comment]
an easy version would just add a tiny avatar next to the current project slug rather than a new column. Is that what y'all were envisioning @benvinegar @EvanPurkhiser?

image

**Comment by EvanPurkhiser** [comment]
Yep! exactly that!

**Comment by benvinegar** [comment]
Yeah that's what I was envisioning. I see your point that it adds clutter in a way the separate column doesn't. I'm all for seeing how this does and coming back to it.

**Comment by benvinegar** [comment]
... Alternatively the whole thing just becomes a column (callsign + platform icon). That way we declutter this area as well as solve my ask that we tie callsign to platform icon.

**Comment by Chrissy** [comment]
I actually think this will work better layout-wise vs adding a new column with both the platformIcon and the slug. I was struggling pretty hard with some of the media queries. 

Let's try this and if it looks super busy we can reevaluate.

**Comment by Chrissy** [comment]
this is rtg and the screenshot is updated. just need a follow-up review!

**Comment by benvinegar** [comment]
I can't speak to the code, but the UI change looks great to me. 👍

File: src/sentry/static/sentry/app/components/autoSelectText.jsx
Comment (lynnagara) [inline src/sentry/static/sentry/app/components/autoSelectText.jsx]: nit: we import this as `classNames` not cx everywhere else, can we match?

**Review by lynnagara — APPROVED** [review]
Looks good apart from the nit about importing classnames library

**Pull request merged**
